src/common/user-store/user-store.ts

```diff
@@ -109,7 +116,29 @@ export class UserStore extends BaseStore<UserStoreModel> /* implements UserStore
       });
     }, {
       fireImmediately: true,
-    });
+    }); 
+  }
+
+  // Returns monaco editor options for selected editor type (the place, where a particular instance of the editor is mounted)
+  getEditorOptions(type: EditorType): monaco.editor.IStandaloneEditorConstructionOptions{
+    return {
+      automaticLayout: true,
+      tabSize: this.editorConfiguration.tabSize ?? defaultEditorConfig.get(type).tabSize,
+      minimap: this.editorConfiguration.miniMap ?? defaultEditorConfig.get(type).miniMap,
+      lineNumbers: this.editorConfiguration.lineNumbers
+    };
+  }
+
+  setLineNumbers(lineNumbers: monaco.editor.LineNumbersType) {
+    this.editorConfiguration.lineNumbers = lineNumbers;
+  }
+
+  setTabSize(tabSize: number) {
+    this.editorConfiguration.tabSize = tabSize;
+  }
+
+  enableMinimap(miniMap: Boolean) {
+    this.editorConfiguration.miniMap = miniMap!= null ? {enabled: <boolean>miniMap} as monaco.editor.IEditorMinimapOptions : null;
```

Comment:
Please don't use `<T>expr` casts. They are dangerous.

Furthermore, you shouldn't need to have a cast in a set config function.